### PR TITLE
Initial attempt at an API refactor 

### PR DESCRIPTION
**Pull request by miguel-vila**
This is not intended to be a final PR and it's just to start a conversation. This is intended to address #52

A summary of some of the changes:

- Simplify `DogStatsDClient` and `StatsDClient` by making them depend on `Client` directly
- Make `Client` depend on an initialized queue
- Encapsulate queue creation and shutdown with constructors that return a `Managed`
- Comment out some constructors for the time being, could be replaced by mapping the `ZManaged`

Some open questions I still have:
- The `.orDie` in the `fiber.join` doesn't look good, not sure how that thread should be cleaned up(?)
- Not sure the shutdown order is right. Think `ClientTest` has been broken.

### CODE REVIEW

File: statsd/src/main/scala/zio/metrics/statsd/StatsDClient.scala [hunk 1]
```diff
@@ -1,104 +1,94 @@
...
 
-  def set(name: String, value: String, sync: Boolean)(implicit queue: Queue[Metric]): Task[Unit] = {
-    val sendM = if (sync) send(queue) else sendAsync(queue)
+  def set(name: String, value: String, sync: Boolean): Task[Unit] = {
+    val sendM = if (sync) client.send else client.sendAsync
     sendM(Set(name, value, Seq.empty[Tag]))
   }
 }
 
 object StatsDClient {
```
Comment (toxicafunk) [inline statsd/src/main/scala/zio/metrics/statsd/StatsDClient.scala:77]: Companion object is apparently not needed anymore.

File: statsd/src/main/scala/zio/metrics/dogstatsd/DogStatsDClient.scala [hunk 2]
```diff
@@ -160,29 +138,29 @@ class DogStatsDClient(
...
     sync: Boolean
-  )(implicit queue: Queue[Metric]): Task[Unit] = {
-    val sendM = if (sync) send(queue) else sendAsync(queue)
+  ): Task[Unit] = {
+    val sendM = if (sync) client.send else client.sendAsync
     sendM(Event(name, text, timestamp, hostname, aggregationKey, priority, sourceTypeName, alertType, tags))
   }
 }
 
 object DogStatsDClient {
```
Comment (toxicafunk) [inline statsd/src/main/scala/zio/metrics/dogstatsd/DogStatsDClient.scala:147]: Not needed anymore it seems.

**Comment by toxicafunk** [comment]
@miguel-vila the tests are fine, the problem is you need to also update the documentation, in particular `docs/essential/statsd.md`.

You can verufy by running `sbt docs/mdoc`

**Comment by miguel-vila** [comment]
hello @toxicafunk 👋 , made some changes:

- 2db78b4: made some stuff private, added some start constructors for the specific clients, I hope those are useful.
- b84b533: updated the docs, this included changing some of the code and also rewording some stuff, I hope I covered everything and that it is clear, let me know how it looks to you and if you want to reword something or change anything go ahead.

**Pull request merged**